src/Libraries/RevitNodes/GeometryConversion/ProtoToRevitFace.cs

```diff
@@ -0,0 +1,154 @@
+using Solid = Autodesk.DesignScript.Geometry.Solid;
+using Surface = Autodesk.DesignScript.Geometry.Surface;
+
+namespace Revit.GeometryConversion
+{
+   [SupressImportIntoVM]
+   public static class ProtoToRevitSolid
+   {
+     /// <summary>
+     /// this method attmempts to construct a BRep from a closed solid.
+     /// </summary>
+     /// <param name="sol"></param>
+     /// <param name="performHostUnitConversion"></param>
+     /// <returns></returns>
+       public static GeometryObject ToRevitType(Solid sol,
+          bool performHostUnitConversion = true)
+       {
+           try
+           {
+               return ToRevitType(sol, performHostUnitConversion, BRepType.Solid);
+           }
+           catch
+           {
+               return null;
+           }
+       }
+
+       /// <summary>
+       /// this method attmempts to construct a BRep from a surface.
+       /// </summary>
+       /// <param name="surf"></param>
+       /// <param name="performHostUnitConversion"></param>
+       /// <returns></returns>
+       public static GeometryObject ToRevitType(Surface surf,
+          bool performHostUnitConversion = true)
+       {
+           try
+           {
+               return ToRevitType(surf, performHostUnitConversion, BRepType.OpenShell);
+           }
+           catch
+           {
+               return null;
+           }
+       }
+
+       private static GeometryObject ToRevitType(Autodesk.DesignScript.Geometry.Topology topology,
+       bool performHostUnitConversion,
+       BRepType type)
+       {
+           List<Face> faces = null;
+
+           if (performHostUnitConversion)
+           {
+               var newTopology = topology.InHostUnits();
+               faces = newTopology.Faces.ToList();
+               newTopology.Dispose();
+           }
+           else
+           {
+               faces = topology.Faces.ToList();
+           }
+
+           var brb = new BRepBuilder(type);
+           var edgeDict = new Dictionary<Edge, BRepBuilderGeometryId>();
+
+           //foreach face in solid
+           foreach (Face protoFace in faces)
+           {
+               var geom = protoFace.SurfaceGeometry();
+
+               var ngeom = geom.ToNurbsSurface();
+               bool flipped = false;
+
+               // Check if the nurbs surface has flipped compared to the original surface
+               if (geom.NormalAtParameter(.5, .5).Dot(ngeom.NormalAtParameter(.5, .5)) < 0)
+               {
+                   flipped = true;
+               }
+
+               // Create Revit nurbs surface
+               var bbface = BRepBuilderSurfaceGeometry.CreateNURBSSurface(ngeom.DegreeU, ngeom.DegreeV,
+                   ngeom.UKnots(), ngeom.VKnots(), ngeom.ControlPoints().SelectMany(x => x.Select(y => y.ToXyz())).ToList(),
+                   ngeom.Weights().SelectMany(x => x).ToList(),
+                   false,
+                   new BoundingBoxUV());
```

Comment:
I remember not being sure about using a default boundingBox here, do we have any more information on what this should be?